registry.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oras
+
+import (
+	"context"
+
+	"oras.land/oras-go/v2/registry"
+)
+
+// Tag fetches the manifest identified by the srcRef, then
+// pushes the manifest with the dstRef
+func Tag(ctx context.Context, target Target, srcRef, dstRef string) error {
```

Comment:
> This method is not efficient as expected for `remote.Repository` since it will request 2 auth tokens instead of one.
> 
> We probably need to implement something like `Repository.TagReference(ctx context.Context, src, dst string) error` so that only 1 auth token is requested.
> 
> @Wwwsylvia What do you think?

@shizhMSFT I'm not very sure about this one. Do you mean adding a new method TagReference into the Respository interface under registry package? And the implementation for remote/Repository would be combining FetchReference and PushReference using 1 token, something like ctx = withScopeHint(ctx, ref, auth.ActionPull, auth.ActionPush)?